### ISSUE TITLE
Partial View Infinite Editor breaks with nested partial

### ISSUE THREAD

**Issue by DanDiplo** [DanDiplo/Umbraco.GodMode#23]
In God Mode there is a partial browser - `/umbraco/#/settings/godModeTree/partialBrowser/browse` - that lets you view all partials. This now comes with an infinite editor, so when you click the partial it opens up in the infinite editor:

![image](https://user-images.githubusercontent.com/6676168/135992901-46c4433e-ef17-4d76-bd3e-b0208e26cc2d.png)

This works fine if the partial is in the root of `/Views/Partials/` but fails (with a 404) if the partial is nested within a folder.  So, in the example in the screenshot, `CategoryLinks.cshtml` and `SectionHeader.cshtml` open OK, but `SubNavigation.cshtml` fails, as it's nested with a sub-folder called `Navigation`.

The infinite editor is in `GodMode.PartialBrowser.Controller.js` and runs the following code to open the partial:

```
vm.openPartial = function (path) {
                const editor = {
                    view: "views/partialViews/edit.html",
                    id: path,
                    submit: function () {
                        vm.init();
                        editorService.close();
                    },
                    close: function () {
                        editorService.close();
                    }
                };
                editorService.open(editor);
            };
```

So it passes the full path of the partial to the editor. The partials are parsed from the Umbraco templates returned by `IFileService` `GetTemplates()` method.

So how can we open nested partials in the infinite editor? What is the trick?

At a low-level the following call is being made to the Umbraco API:

`https://localhost:44349/umbraco/backoffice/umbracoapi/codefile/GetByPath?type=partialViews&virtualPath=Navigation%25252FSubNavigation.cshtml`

This is what returns the 404 when the path is nested.

**Comment by Lantzify** [comment]
I am having a bit of a problem getting the project to start. So I might be far off here. 

But couldn't you do something like this in `PartialHelper.cs `
` partial.Path = $"/Views/Partials/{partial.Name }.cshtml";` since partial.Name becomes `Navigation/SubNavigation`.

Right now the path becomes `Navigation/SubNavigation.cshtml`. And since it doesn't have ` /Views/Partials`, Umbraco won't find the partial view?

Or am I thinking completely wrong here?

**Comment by DanDiplo** [comment]
@Lantzify I'll give that a try and see if it works - thanks for the suggestion! If you need any help getting it running, let me know, too.

**Comment by Lantzify** [comment]
Okay @DanDiplo! I got it up and running! And here are my findings!

And it looks like the "~" is the problem here for some reason.. I did this
```
partial.Path = partial.Name.Replace("~", string.Empty).Replace("/", "%252F").Replace("~%252FViews%252FPartials%252F", string.Empty);
```
This loads the nested partial views. But the path becomes:
```
 /Views/Partials/Views/Partials/Navigation/SubNavigation.cshtml
```
And this introduces a new error. If you try to save the file, it dosen't register. And the overlay "unsaved changes" triggers.

I tried adding the `IWebHostEnvironment` like such:
```
 partial.Path = webHostEnvironment.ContentRootFileProvider.GetFileInfo(partial.Name.Replace("~", string.Empty)).PhysicalPath;
                    partials.Add(partial);
```
And this seems to work fine. BUT the path loaded becomes according to the image. But after saving. The path becomes normal. 
` ¯\_(ツ)_/¯`

I pushed the changed to [my fork](https://github.com/Lantzify/Umbraco.GodMode/commit/125858e9026a7f39a145f717f2bc40e480f820b9). Maybe a step in the right direction?

![Capture](https://user-images.githubusercontent.com/20320733/137590609-d882e937-ddd6-4afd-bcb9-41628dac722d.PNG)

**Comment by DanDiplo** [comment]
Nice work! I'm going to play around with this and see where it gets us... thanks for your time and effort, it's appreciated!

**Comment by DanDiplo** [comment]
@Lantzify OK, so I managed to fix this! Whilst I didn't use your solution directly, it did give me some ideas that helped me track the issue down. Your solution relied on passing the full path, which did work, but as you noticed caused issues with saving. I went back to looking at using the relative path as I examined the Umbraco source code and could see this was the normal way of doing so.

Essentially it boiled down to an encoding problem - I was encoding the file path in the C# code, but this wasn't quite correct, so I ended up encoding it in the JS using `encodeURIComponent()`.  

So I just need the unencoded relative path from the `/Views/Partials/` folder in my helper and then encoded it in the Angular.
